tests/mcp/test_mcp_js_exec.py

```
@@ -0,0 +1,171 @@
+    url = "https://httpbin.org/html"
+
+    # Scripts that should return actual values
+    scripts = [
+        "return document.title || 'No title found'",  # Handle empty title case
+        "return document.querySelectorAll('h1').length",
+        "return Array.from(document.querySelectorAll('p')).map(p => p.textContent.substring(0, 50))",
+        "return 'test string'",  # Simple string test
+        "return 42"  # Simple number test
+    ]
+
+    response = requests.post("http://localhost:11235/execute_js", json={
```

Ruff (0.13.1)

Probable use of `requests` call without timeout

(S113)